### ISSUE TITLE
SoftmaxMultiClassObj: label exceed num_class 

### ISSUE THREAD

**Issue by arshak** [antinucleon/XGBoost.jl#4]
I am using covertype 

unique(labels)
7-element Array{Int64,1}:
 2
 1
 5
 6
 7
 3
 4

bst = xgboost(instances, num_round, label = labels, eta=1, max_depth=10, num_class=7, objective="multi:softmax")

errors out with SoftmaxMultiClassObj: label exceed num_class 

however 

bst = xgboost(instances, num_round, label = labels, eta=1, max_depth=10, num_class=8, objective="multi:softmax")

trains a model.

**Comment by tqchen** [comment]
The multi-class label should range from [0, 6]

**Comment by arshak** [comment]
Got it however the code shouldn't just crash when this happens.  Right now my Julia just exists after the error.

**Comment by antinucleon** [comment]
I am sorry I didn't update it for a long time but at the same time xgboost
changes a lot. I hope I will have time next week to fully upgrade it.

**Comment by arshak** [comment]
Excellent, thanks for the fast responses!
